### PR TITLE
Add app password migration section

### PR DESCRIPTION
**Pull request by rullzer**
Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: developer_manual/client_apis/LoginFlow/index.rst [hunk 1]
```diff
@@ -54,3 +54,17 @@ This information will be used by the client to create a new account.
 After this the webview is destroyed including all the state the webview holds.
 
 .. note:: On Nextcloud 12 the returned server is just the server address without any possible subfolder. This is corrected in Nextcloud 13.
+
+
+Converting to app passwords
+---------------------------
+
+Old configurations of clients might still be using username and passwords. The login flow ensures that each device has a unique app password.
```
Comment (tobiasKaminsky) [inline developer_manual/client_apis/LoginFlow/index.rst]:
```suggestion
Old configurations of clients might still be using username and passwords. The login flow ensures that each device has an unique app password.
```

File: developer_manual/client_apis/LoginFlow/index.rst [hunk 2]
```diff
@@ -54,3 +54,17 @@ This information will be used by the client to create a new account.
...
+In order to facilitate transparent migration to app passwords there is an endpoint that can be called by client.
+
+If the client is authenticated with an app password a 403 will be returned. If the client is authenticating with a real password an app password will
+be generated and returned.
```
Comment (tobiasKaminsky) [inline developer_manual/client_apis/LoginFlow/index.rst]: in one line?

File: developer_manual/client_apis/LoginFlow/index.rst [hunk 3]
```diff
@@ -54,3 +54,17 @@ This information will be used by the client to create a new account.
...
+
+.. code-block:: bash
+
+     curl -u username:password -H 'OCS-APIRequest: true' https://cloud.example.com/ocs/v2.php/core/getapppassword
```
Comment (tobiasKaminsky) [inline developer_manual/client_apis/LoginFlow/index.rst]: what is the output if a new app password is sent back?
Just plain app token? Would be good to have this response included.

Comment (tobiasKaminsky) [reply developer_manual/client_apis/LoginFlow/index.rst]:
```
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>ok</status>
  <statuscode>200</statuscode>
  <message>OK</message>
 </meta>
 <data>
  <apppassword>M1DqHwuZWwjEC3ku7gJsspR7bZXopwf01kj0XGppYVzEkGtbZBRaXlOUxFZdbgJ6Zk9OwG9x</apppassword>
 </data>
</ocs>

```

**Comment by tobiasKaminsky** [comment]
Can we also pass an identifier?
Or is this done automatically? 
(it would be good that somehow the client e.g. Android client with Device model can be used)

**Comment by MorrisJobke** [comment]
What is the status here?

**Comment by MorrisJobke** [comment]
> What is the status here?

🏓

**Comment by rullzer** [comment]
> What is the status here?

me needing to finish it

**Comment by MorrisJobke** [comment]
> me needing to finish it

Ready now?

**Pull request merged**